airflow/contrib/auth/backends/google_auth.py

```diff
@@ -144,7 +145,11 @@ def load_user(self, userid):
     def oauth_callback(self):
         _log.debug('Google OAuth callback called')
 
-        next_url = request.args.get('next') or url_for('admin.index')
+        state = json.loads(request.args.get('state'))
+        try:
+            next_url = state.get('next')[0]
```

Comment:
I think blind redirect to a querystring param without validating that the domain is trusted is considered a security vulnerability:
https://security.stackexchange.com/questions/121643/is-it-safe-to-redirect-to-url-parameter-without-filtering

Reply:
Yeah, good point. I'll [encode the state using JWT](https://tools.ietf.org/html/draft-bradley-oauth-jwt-encoded-state-05), so we can verify that it hasn't been tampered with.